activerecord/lib/active_record/associations/association.rb

```diff
@@ -29,16 +18,50 @@ module Associations
     #
     #   blog = Blog.first
     #
-    # The association of <tt>blog.posts</tt> has the object +blog+ as its
-    # <tt>owner</tt>, the collection of its posts as <tt>target</tt>, and
-    # the <tt>reflection</tt> object represents a <tt>:has_many</tt> macro.
-    class Association # :nodoc:
-      attr_accessor :owner
-      attr_reader :reflection, :disable_joins
+    # The association of +blog.posts+ has the object +blog+ as its +owner+, the
+    # collection of its posts as +target+, and the +reflection+ object
+    # represents a +:has_many+ macro.
+    #
+    #--
+    #
+    # The following represents the inheritance chain for association classes,
+    # with Association at the root:
+    #
+    #
+    # * Association
+    #   * SingularAssociation
+    #     * HasOneAssociation includes ForeignAssociation
+    #       * HasOneThroughAssociation includes ThroughAssociation
+    #     * BelongsToAssociation
+    #       * BelongsToPolymorphicAssociation
+    #   * CollectionAssociation
+    #     * HasManyAssociation includes ForeignAssociation
+    #       * HasManyThroughAssociation includes ThroughAssociation
+    #
+
+    class Association
+      attr_accessor :owner # :nodoc:
+      attr_reader :reflection, :disable_joins # :nodoc:
 
       delegate :options, to: :reflection
 
-      def initialize(owner, reflection)
+      ##
+      # :attr_reader: owner
```

Comment:
Should this be an accessor?
```suggestion
      # :attr_accessor: owner
```